src/corporate-actions/corporate-actions.service.ts

```diff
@@ -75,35 +73,29 @@ export class CorporateActionsService {
     ticker: string,
     dividendDistributionDto: DividendDistributionDto
   ): ServiceReturn<DividendDistribution> {
-    const { signer, webhookUrl, originPortfolio, ...rest } = dividendDistributionDto;
+    const { base, args } = extractTxBase(dividendDistributionDto);
```

Comment:
```suggestion
    const { base, args: {originPortfolio, ...rest} } = extractTxBase(dividendDistributionDto);
```
If you could have done like this, the below code wouldn't have changed